### ISSUE TITLE
Cannot dynamically update FileName

### ISSUE THREAD

**Issue by mazz** [igor11191708/swiftui-loop-videoPlayer#4]
If you use a variable in the `Filename(myVar)` it will not update. For example:

```swift
struct WalkthroughPageView<ActionView: View>: View {
    @Binding var intro: WalkthroughPage
    var body: some View {
        VStack {
          LoopPlayerView {
                  Settings {
                      FileName(intro.introAssetName)
                      Ext("mp4")
                      Gravity(.resizeAspect)
                  }
              }
```

The video that appears is from the first string passed-in(never changes)

**Comment by swiftuiux** [comment]
Hi
Updated, try it
![change_video_file](https://github.com/user-attachments/assets/f2c714a6-f927-434b-afd9-3e5859f05a72)

**Comment by mazz** [comment]
@The-Igor with your latest changes I still see the issue. Here is an example project where the video does not get updated. 

Click Next at the bottom of the page to have it cycle to the next page. The video should change from birds, to cars, to fruit.

If you revert the commented-out lines you will see the code update the `Image` but when you use `LoopPlayerView` the video does not update.

[WalkthroughPageAnimation.zip](https://github.com/user-attachments/files/16474215/WalkthroughPageAnimation.zip)

**Comment by swiftuiux** [comment]
ok I'll take a look next week

**Comment by swiftuiux** [comment]
I checked the code
I see that the project uses release 1.4.1
All last changes were not released they are in the main brunch I thought that you would try the main brunch
I changed the source of the package I think everything works as expected Check video and try one more time
I released 1.4.2 with all changes

![swiftui](https://github.com/user-attachments/assets/a31e9cb0-4334-4838-9c2e-baa99db908eb)

**Comment by mazz** [comment]
1.4.2 seems to work great! Thanks!
